Game Design Principles Update (#1650)
PixelGuys/Cubyz: GAME_DESIGN_PRINCIPLES.md

# Game Design Principles
This document is intended for contributors and may contain spoilers.

## What is Cubyz?

> Cubyz is a voxel sandbox game that aims to take advantage of the infinite world it provides. The player has free agency to do whatever they please, though their efforts will always be challenged.

Reviewer: ```suggestion
Cubyz is a voxel sandbox game with infinite world with focus on great adventures, immersive exploration and numerous challenges along the way. We cherish freedom and we are determined to give it to the players in abundance, so they can chose their own path through the world.
```

### No Automation
Having quick, infinite resources at the palm of players' hands discourages exploration, as the player will never need to forage or search for blocks they desire.

### Mobs Don't Respawn Naturally
Clearing a dangerous area of its monsters will make it safe to build and explore.
This rule is also here to prevent mob farming.

### No Passive Animals
Animals do not want to die, so they will either run from an attacking player or attempt to defend themselves. The player will have to strategize to be able to hunt.

### No Unbreakable Tools
If a player gets too attached to their amazing tool, they won't want to make other types of tools, and low-tier materials will see less use as they will never be used in tool-making.

### Avoid Clutter
Inventories will often fill up with random items that the player does not want, making their inventory hard to manage. To mitigate this, find uses for existing items before adding new ones and prevent items from finding their way into a player's inventory when they don't want it.

### Use Realism Tastefully

> Cubyz is not a realistic game. The world is very whimsical and has rules of its own, sometimes inspired by the real world, but never copying it outright.

Reviewer 2: I think sometimes it might make sense to copy real world mechanics, but there should be a good reason for it.
```suggestion
Cubyz is not a realistic game. The world is very whimsical and has rules of its own, sometimes inspired by the real world, but never copied without a good reason.
```

- Find an above-ground structure that can grant the player coal.
- Dig straight down until you find coal.

This problem is completely solved as soon as the player gets coal, as they can now explore caves to find more coal with their newfound torches. It's important that the player has these options to prevent them from getting stuck on a seemingly insurmountable barrier. This gives them more freedom in how they can approach the game, as no one option is particularly the best and can change depending on situation and location.

### Explain when Needed
Progress-related mechanics need explanations so the players don't get stuck. Keep it simple on the surface so it's easily approachable, but leave out details that the player can find themselves through intuition.

### Show, Don't Tell

> An important rule, showing how a mechanic works instead of telling encourages player experimentation and reduces confusion. Here are some examples:

Reviewer 2: I don't think we can really do this. Some things cannot be discovered at all (e.g. gem encasing in tool crafting) and some things are better told up front, we don't want the player to die from food poisoning, just because no one warned them. This is the kind of stuff that makes you check the wiki before eating new food types.
